Doc/library/pathlib.rst

```diff
@@ -1639,6 +1611,61 @@ call fails (for example because the path doesn't exist).
    .. versionchanged:: 3.10
       The *newline* parameter was added.
 
+
+.. _pattern-language:
+
+Pattern language
+----------------
+
+The following wildcards are supported in patterns for
+:meth:`~PurePath.full_match`, :meth:`~Path.glob` and :meth:`~Path.rglob`:
+
+=========================  ===========================================
+Wildcard                   Matches
+=========================  ===========================================
+"``**``" (full segment)    Any number of file or directory segments.
+"``*``" (full segment)     One file or directory segment.
+"``*``" (otherwise)        Any number of non-separator characters.
+"``?``"                    One non-separator character.
+"``[seq]``"                One character in *seq*.
+"``[!seq]``"               One character not in *seq*.
+=========================  ===========================================
+
+For a literal match, wrap the meta-characters in brackets.
+For example, ``"[?]"`` matches the character ``"?"``.
+
+The "``**``" wildcard enables recursive globbing. A few examples:
+
+=========================  ===========================================
+Pattern                    Meaning
+=========================  ===========================================
+"``**/*``"                 Any path with at least one segment.
+"``**/*.py``"              Any path with a final segment ending "``.py``".
+"``assets/**``"            Any path starting with "``assets/``".
+"``assets/**/*``"          Any path starting with "``assets/``", excluding "``assets/``" itself.
+=========================  ===========================================
+
+.. note::
+   Globbing with the "``**``" wildcard visits every directory in the tree.
+   Large directory trees may take a long time to search.
+
+.. versionchanged:: 3.13
+   Globbing with a pattern that ends with "``**``" returns both files and
+   directories. In previous versions, only directories were returned.
+
+In :meth:`Path.glob` and :meth:`~Path.rglob`, a trailing slash may be added to
+the pattern to match only directories.
+
+.. note::
+   This module normally removes trailing slashes. Paths returned from
+   :meth:`Path.glob` and :meth:`~Path.rglob` don't include any trailing slash
```

Comment:
Isn't it more accurate to say that `Path` objects don't keep a trailing slash? And `Path.[r]glob` return `Path` objects.

Maybe rather than a scary note, we could just add to the previous paragraph (line 1657): `Remember that directories returned as ``Path`` objects do not have a trailing slash.`?

Reply:
Any better? I removed the "remember that" formulation as it seemed a bit unfair to expect users to remember something from near the beginning of the pathlib docs (this section is towards the end)